src/printer.rs

```diff
@@ -771,31 +746,39 @@ fn collect_interned_values<'tcx>(tcx: TyCtxt<'tcx>, items: Vec<&MonoItem>) -> In
     for item in items.iter() {
         match &item {
             MonoItem::Fn(inst) => {
-                for body in get_bodies(tcx, inst).into_iter() {
-                    InternedValueCollector {
-                        tcx,
-                        _sym: inst.mangled_name(),
-                        locals: body.locals(),
-                        link_map: &mut calls_map,
-                        visited_tys: &mut visited_tys,
-                        visited_allocs: &mut visited_allocs,
-                    }
-                    .visit_body(&body)
+                let body = inst.body().unwrap_or_else(|| {
+                    panic!(
+                        "Failed to retrive body for Instance of MonoItem::Fn {}",
+                        inst.name()
+                    )
+                });
```

Comment:
So this looks like the cases you found where `MonoItem:Fn` does not have a body are somehow filtered out before? Or would we rather want to just report it (or not even that) and move on when there is no `body` to visit?